### PR TITLE
transfer to eywalker

### PR DESCRIPTION
**Pull request by fabiansinz**

### CODE REVIEW

File: attorch/layers.py [hunk 1]
```diff
@@ -86,296 +87,236 @@ def forward(self, input):
...
-            r += '  -> ' + ch.__repr__() + '\n'
-        return r
+        return ('spatial positive ' if self.positive else '') + \
+               ('normalized ' if self.normalize else '') + \
+               self.__class__.__name__ + \
+               ' (' + '{} x {} x {}'.format(*self.in_shape) + ' -> ' + str(
+            self.outdims) + ')'
 
 
-class SpatialTransformerLaplace2d(nn.Module):
```
Comment (eywalker) [inline attorch/layers.py]: Why did this get removed? This will cause some code to break.

File: attorch/regularizers.py [hunk 2]
```diff
@@ -3,10 +3,13 @@
...
 import torch
 from itertools import product
+from torch.nn import functional as F
 
+# def laplace():
+#     return np.array([[0.25, 0.5, 0.25], [0.5, -3.0, 0.5], [0.25, 0.5, 0.25]]).astype(np.float32)[None, None, ...]
 
 def laplace():
-    return np.array([[0.25, 0.5, 0.25], [0.5, -3.0, 0.5], [0.25, 0.5, 0.25]]).astype(np.float32)[None, None, ...]
+    return np.array([[0,-1, 0], [-1, 4, -1], [0, -1, 0]]).astype(np.float32)[None, None, ...]
```
Comment (eywalker) [inline attorch/regularizers.py]: I'll be concerned that this change will make existing trained networks invalid. What's the basis for changing the smoothening filter?

Comment (fabiansinz) [reply attorch/regularizers.py]: It won't make them invalid. They just won't train to the same state with the same parameters. I changed it to make it consistent with the filter that Alex uses.

File: attorch/layers.py (hunk 1 above)
Comment (fabiansinz) [reply attorch/layers.py]: Do you use that layer? I don't, but I can put it back.

**Pull request merged**